modal/sandbox.py

```diff
@@ -710,10 +752,31 @@ async def _load(self: _SandboxSnapshot, resolver: Resolver, existing_object_id:
         return obj
 
     @staticmethod
-    async def _experimental_from_snapshot(snapshot: _SandboxSnapshot, client: Optional[_Client] = None):
+    async def _experimental_from_snapshot(
+        snapshot: _SandboxSnapshot,
+        client: Optional[_Client] = None,
+        *,
+        name: Optional[Union[str, type[DefaultSandboxNameOverride]]] = _DEFAULT_SANDBOX_NAME_OVERRIDE,
```

Comment:
I wonder if making your `DefaultSandboxNameOverride` type subclass `str` would let you avoid this ugly annotation.